Cython/Compiler/FusedNode.py

```diff
@@ -687,30 +711,65 @@ def __pyx_fused_cpdef(signatures, args, kwargs, defaults):
             env.use_utility_code(Code.UtilityCode.load_cached("Import", "ImportExport.c"))
             env.use_utility_code(Code.UtilityCode.load_cached("ImportNumPyArray", "ImportExport.c"))
 
+        context.update(
+            global_sigindex_name = fused_cpdef_globalindex,
+            func_sigindex_key = env.mangle(fused_cpdef_globalindex_prefix, orig_py_func.entry.cname)
+        )
+
+        global_scope = env.global_scope()
+        if not global_scope.lookup_here(fused_cpdef_globalindex):
+            from .Builtin import dict_type
+            global_scope.declare_var(StringEncoding.EncodedString(fused_cpdef_globalindex), dict_type, None, is_cdef=True)
```

Comment:
Pushing this through a global user-visible declaration doesn't seem ideal. I rather liked your solution with the local default argument, because it's really the most self-contained. I'd rather see the bug resolved that prevents that from working, than leaking more than necessary into the global user namespace. My intuition says that it might be easy to fix, could just be a missing attribute traversal in a transform somewhere (which wasn't relevant before but becomes necessary now), or maybe copying something more over from the `TreeFragment`. I can see if I can figure something out.

Reply:
I personally felt like the original local default argument implementation was a bit hack-y, since it used arguments for something other than receiving information from outside the function, it relied on the default value being an object persistent across separate calls (which feels a bit like an incidental language/implementation detail), and it could in theory be broken by a call that overrides the default argument value. In practice, none of those concerns should ever lead to any real issues though, and I think it might have been a tiny bit faster too (though I'm not sure why), so yeah, it's probably still better than having the indices be user-visible.

I'll move the indices back to default arguments once it no longer segfaults.